src/main/java/com/fakng/fakngagrgtr/parser/apple/AppleParser.java

```diff
@@ -0,0 +1,145 @@
+        return Integer.parseInt(doc.select(COUNT_PAGES_ADDRESS).get(1).text());
+    }
+
+    private List<Vacancy> takeVacanciesFromDoc(Document doc) throws ParseException {
+        String jsonBody = takeJsonBody(doc);
+        return takeVacanciesFromJsonBody(jsonBody);
+    }
+
+    private String takeJsonBody(Document doc) {
+        return doc.select(JSON_BODY_ADDRESS)
+                .first()
+                .childNode(0).toString()
+                .replace(";", "")
+                .trim()
+                .substring(19);
+    }
+
+    private List<Vacancy> takeVacanciesFromJsonBody(String json) throws ParseException {
+        JSONObject jsonBody = parse(json);
+        JSONArray searchResults = (JSONArray) jsonBody.get("searchResults");
+        List<Vacancy> vacancies = new ArrayList<>();
+        for (Object vacancyJson : searchResults) {
+            try {
+                VacancyApple vacancyApple = toVacancyApple(vacancyJson);
+                vacancies.add(toVacancy(vacancyApple));
+            } catch (JsonProcessingException e) {
+                e.printStackTrace();
+            }
+        }
+        return vacancies;
+    }
+
+    private JSONObject parse(String json) throws ParseException {
+        JSONParser jsonParser = new JSONParser(JSONParser.MODE_JSON_SIMPLE);
+        return  (JSONObject) jsonParser.parse(json);
+    }
+
+    private VacancyApple toVacancyApple(Object vacancyJson) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        return mapper.readValue(vacancyJson.toString(), VacancyApple.class);
+    }
+
+    private Vacancy toVacancy(VacancyApple vacancyApple) {
+        Vacancy vacancy = new Vacancy();
+        vacancy.setTitle(vacancyApple.getPostingTitle());
+        vacancy.setDescription(generateFullDescription(vacancyApple));
+        vacancy.setUrl(URL_FOR_VACANCY + vacancyApple.getPositionId());
+        vacancy.setAddDate(LocalDateTime.now());
+        vacancy.setPublishedDate(parseVacancyAppleDateTime(vacancyApple.getPostDateInGMT()));
+        vacancy.setCompany(company);
+        vacancy.setLocations(toLocations(vacancyApple.getLocations()));
+        return vacancy;
+    }
+
+    private String generateFullDescription(VacancyApple vacancyApple) {
+        return vacancyApple.getTeam().getTeamName() + "\n" +
+                vacancyApple.getPostingTitle() + "\n" +
+                vacancyApple.getJobSummary();
+    }
+
+    private LocalDateTime parseVacancyAppleDateTime(String dateTime) {
+        return LocalDateTime.parse(dateTime.replace("Z", ""));
+    }
+
+    private List<Location> toLocations(List<LocationApple> locationsApple) {
+        return locationsApple.stream()
+                .map(location -> locationProcessor.processLocation(company, location.getCity(), parseCountry(location.getCountryID())))
+                .collect(Collectors.toList());
+    }
+
+    private String parseCountry(String country) {
+        return country.split("-")[1];
```

Comment:
I'd rather use country.substring(country.indexOf('-'), country.length())
Because it doesn't create a new array inside

But it's applicable only if this is your country string format:
{part1}-{part2}
And you're willing to get part2